### PR TITLE
Rewrite get federation deliveryservices to Go

### LINKED ISSUE #3793: Rewrite /federations/{{id}}/deliveryservices to Go
## I'm submitting an ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/federations/{{id}}/deliveryservices` is handled by Perl

## Expected / new behavior:
`/federations/{{id}}/deliveryservices` should be handled by Go.

### Minimum API Version
1.1

### Request Methods
- [ ] `GET`
- [x] `POST` (#2862)

## Minimal reproduction of the problem with instructions:
N/A

### PR DESCRIPTION
**Pull request by mhoppa**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3793

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Build TO with change locally and use the GET federations/{id}/deliveryservices endpoint. It should behave the exact way it did before. Another way to verify it is to run the API tests.

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: docs/source/api/federations_id_deliveryservices.rst [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 ``GET``
 =======
-Retrieves :term:`Delivery Service`\ s assigned to a federation.
+Retrieves :term:`Delivery Service`\ s assigned to a :term:`Federation`.
```
Comment (ocket8888) [inline docs/source/api/federations_id_deliveryservices.rst]: Not necessary, but I'd appreciate it if while you're doing this if you could get rid of the `\ s`; `:term:` can handle plurals now.

File: traffic_ops/client/federation.go [hunk 2]
```diff
@@ -78,3 +79,12 @@ func (to *Session) CreateFederationDeliveryServices(federationID int, deliverySe
 	inf, err := makeReq(to, http.MethodPost, apiBase+`/federations/`+strconv.Itoa(federationID)+`/deliveryservices`, jsonReq, &resp)
 	return inf, err
 }
+
+func (to *Session) GetFederationDeliveryServices(federationID int) ([]tc.FederationDeliveryServiceNullable, ReqInf, error) {
```
Comment (ocket8888) [inline traffic_ops/client/federation.go]: Client functions should have GoDocs IMO. Not as big a deal on other things, but this is directly exposed as a part of an interface to our system.

File: traffic_ops/testing/api/v14/federations_test.go [hunk 3]
```diff
@@ -93,4 +93,16 @@ func PostTestFederationsDeliveryServices(t *testing.T) {
 	if _, err = TOSession.CreateFederationDeliveryServices(fedID, []int{ds.ID}, true); err != nil {
 		t.Errorf("creating federations delivery services: %v\n", err)
 	}
+
+	// Test get created Federation Delivery Service
+	fedDSes, _, err := TOSession.GetFederationDeliveryServices(fedID)
+	if err != nil {
+		t.Errorf("cannot GET Federation DeliveryServices: %v\n", err)
```
Comment (ocket8888) [inline traffic_ops/testing/api/v14/federations_test.go]: Should be a `t.Fatalf` - the caller can't safely assume that `fedDSes` in this case isn't `nil`.

I know that the way you wrote that this isn't exactly possible, but in general it's true. I guess I don't really have a problem with that, provided the GoDoc notes that.

File: traffic_ops/traffic_ops_golang/federations/ds.go [hunk 4]
```diff
@@ -104,3 +105,37 @@ func getFedNameByID(tx *sql.Tx, id int) (string, bool, error) {
...
+	return map[string]dbhelpers.WhereColumnInfo{
+		"id":   dbhelpers.WhereColumnInfo{"fds.federation", api.IsInt},
+		"dsID": dbhelpers.WhereColumnInfo{"fds.deliveryservice", api.IsInt},
+	}
+}
+func (v *TOFedDSes) GetType() string {
+	return "federation_deliveryservice"
+}
+
+func (v *TOFedDSes) Read() ([]interface{}, error, error, int) { return api.GenericRead(v) }
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/federations/ds.go]: I don't believe the Perl ever considered tenancy, so  won't demand that you do in the rewrite, but I wanted to point out that you could, if you so chose.

Comment (mhoppa) [reply traffic_ops/traffic_ops_golang/federations/ds.go]: Do you think it is better as part of the port from perl to golang? Or as part of a different effort to add tenancy?

Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/federations/ds.go]: Either way, as you please.

Comment (mhoppa) [reply traffic_ops/traffic_ops_golang/federations/ds.go]: Ill defer to another PR

**Pull request merged**